web/yo/app/scripts/directives/reviewpanel.js

```diff
@@ -144,101 +149,165 @@ angular.module('oncokbApp')
+                                uuid = $scope.tt.name_uuid;
                             }
                             break;
                         case 'CLINICAL_TRIAL':
-                            $scope.tm.trials.clear();
-                            $scope.tm.trials.pushAll($scope.tm.trials_review.get('lastReviewed'));
-                            rejectItem([{reviewObj: $scope.tm.trials_review, uuid: $scope.tm.trials_uuid}], $scope.tm.trials_review);
+                            uuid = $scope.tm.trials_uuid;
                             break;
                         case 'MUTATION_NAME_CHANGE':
-                            rejectItem([{obj: $scope.mt.name, reviewObj: $scope.mt.name_review, uuid: $scope.mt.name_uuid}], $scope.mt.name_review);
-                            break;
                         case 'TUMOR_NAME_CHANGE':
-                            var lastReviewed = $rootScope.model.createList();
-                            _.each($scope.tm.cancerTypes_review.get('lastReviewed'), function(ct) {
-                                var cancerType = $rootScope.model.create(OncoKB.CancerType);
-                                cancerType.cancerType.setText(ct.cancerType);
-                                cancerType.subtype.setText(ct.subtype);
-                                cancerType.cancerType_eStatus.set('obsolete', 'false');
-                                cancerType.subtype_eStatus.set('obsolete', 'false');
-                                cancerType.oncoTreeCode_eStatus.set('obsolete', 'false');
-                                lastReviewed.push(cancerType);
-                            });
-                            $scope.tm.cancerTypes = lastReviewed;
-                            rejectItem([{reviewObj: $scope.tm.cancerTypes_review, uuid: $scope.tm.name_uuid}], $scope.tm.cancerTypes_review);
-                            break;
                         case 'TREATMENT_NAME_CHANGE':
-                            rejectItem([{obj: $scope.tt.name, reviewObj: $scope.tt.name_review, uuid: $scope.tt.name_uuid}], $scope.tt.name_review);
+                            uuid = '';
                             break;
                         default:
                             break;
                         }
+                        if(uuid) {
+                            uuid = uuid.getText();
+                            DatabaseConnector.getEvidencesByUUIDs([uuid], function(result) {
+                                var data = JSON.parse(result.status).data;
+                                if(data.length > 0) {
+                                    $scope.lastUpdateTime = data[0].lastEdit;
+                                }
+                                updateRejectModel();
+                            });
+                        } else {
+                            updateRejectModel();
+                        }
                     });
                 };
+                function updateRejectModel() {
```

Comment:
Why do another swtich-case in here? This should be combined with the swtich-case above